acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/MetadataFileHelpers.java

```diff
@@ -179,6 +184,49 @@ public void createAzureKeyYamlFileAt(
     }
   }
 
+  public void createYubiHsmYamlFileAt(
+      final Path metadataFilePath,
+      final Path destinationYubiShellSimulator,
+      final KeyType keyType) {
+    final int opaqueObjId = keyType == KeyType.BLS ? 1 : 2;
+
+    final Map<String, Serializable> yaml =
+        Map.of(
+            "type",
+            "yubihsm2",
+            "yubiShellBinaryPath",
+            destinationYubiShellSimulator.toString(),
+            "connectorUrl",
+            "http://localhost:12345",
+            "authKey",
+            1,
+            "password",
+            "password",
+            "opaqueObjId",
+            opaqueObjId,
+            "keyType",
+            keyType.name());
+
+    createYamlFile(metadataFilePath, yaml);
+  }
+
+  public static Path copyYubiHsmSimulator(final Path destDir) throws IOException {
```

Comment:
without going into it - this feels quite "hacky" i.e. copying files, injecting shebang, chmod ... is there any other way of doing this?

Reply:
The other way is to either have a shell file (which can work on both mac and linux) or a native go binary for mac and linux.